debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/ChainedReader.java

```diff
@@ -120,13 +120,8 @@ public State state() {
         // We have to be prepared for the current reader to be in the middle of stopping and this chain transitioning
         // to the next reader. In this case, the reader that is stopping will return null or empty, and we can ignore
         // those until the next reader has started ...
-        while (running.get() || !completed.get()) {
-            Reader reader = currentReader.get();
-            if (reader != null) {
-                List<SourceRecord> records = reader.poll();
-                if (records != null && !records.isEmpty()) return records;
-                // otherwise, we'll go ahead until the next reader is ready or until we're no longer running ...
-            }
+        if (running.get() || !completed.get()) {
```

Comment:
Why is this changing? It's behaving quite differently from what's stated above (so at least the comment needs updating, too, if this is actually needed).

Reply:
I'll fix the comment. In original form it worked also in "blocking" manner. It has not returned to poll() if reader has not provided any data.

Reply:
Mh, ok. And what about the possiblity of `currentReader.get()` returning null? This was guarded against before.

Reply:
Yes, you are right, I've oversimplified the code.